Timezone Test Bug part 2: Date Permalinks

### Issue Summary

So in a similar vein to #6975, there is another routes test bug that I keep running into that's related to timezones. This one is related to date permalinks.

```
Frontend Routing Date permalinks should serve RSS with date permalink:
     Uncaught AssertionError: expected -1 to be above 0
      at Assertion.fail (node_modules/should/cjs/should.js:326:17)
      at Assertion.Object.defineProperty.value (node_modules/should/cjs/should.js:398:19)
      at Test.<anonymous> (core/test/functional/routes/frontend_spec.js:562:57)
      at Test.assert (node_modules/supertest/lib/test.js:156:6)
      at Server.assert (node_modules/supertest/lib/test.js:127:12)
      at net.js:1409:10
```

Which is occurring [here](https://github.com/TryGhost/Ghost/blob/master/core/test/functional/routes/frontend_spec.js#L540)

### Steps to Reproduce

1. Set up ghost master
2. Set your timezone to something less than UTC (I'm UTC -7 at the moment)
3. Run `grunt validate` or `grunt test-routes`

Tests should pass but don't.

### Technical details:

* Ghost Version: master (commit ref: https://github.com/TryGhost/Ghost/commit/d5e1d122acc4d8541a666bc36a873d8ea0c50d3f)
* Node Version: 0.12.2 (but someone also said they had it using node 6 -> #7043)
* Browser/OS: Chrome
* Database: SQLite

--- a/core/test/functional/routes/frontend_spec.js
+++ b/core/test/functional/routes/frontend_spec.js
@@ -553,14 +553,13 @@ describe('Frontend Routing', function () {
                     should.exist(res.headers.date);
 
                     var content = res.text,
-                        today = new Date(),
-                        dd = ('0' + today.getDate()).slice(-2),
-                        mm = ('0' + (today.getMonth() + 1)).slice(-2),
-                        yyyy = today.getFullYear(),
+                        todayMoment = moment(),
+                        dd = todayMoment.format('DD'),
+                        mm = todayMoment.format('MM'),
+                        yyyy = todayMoment.format('YYYY'),
                         postLink = '/' + yyyy + '/' + mm + '/' + dd + '/welcome-to-ghost/';
 
                     content.indexOf(postLink).should.be.above(0);
-
                     done();
                 });
         });

